## *USA CAR Dataset Analysis*

##### Objective :
*Analyze the data on USA Cars to find the insights*

##### Source:
*Source CAR dataset is taken from Kaggle*

### *Reading and Inspection*
*Read the data as dataframe and inspect it using Pandas*

In [58]:
import pandas as pd
import numpy as np

car_df = pd.read_csv('/home/dharani/kaggle_input/USA_cars_datasets.csv')
car_df.head(5)

Unnamed: 0  price      brand    model  year   title_status   mileage  \
0           0   6300     toyota  cruiser  2008  clean vehicle  274117.0   
1           1   2899       ford       se  2011  clean vehicle  190552.0   
2           2   5350      dodge      mpv  2018  clean vehicle   39590.0   
3           3  25000       ford     door  2014  clean vehicle   64146.0   
4           4  27700  chevrolet     1500  2018  clean vehicle    6654.0   

    color                  vin        lot       state country      condition  
0   black    jtezu11f88k007763  159348797  new jersey     usa   10 days left  
1  silver    2fmdk3gc4bbb02217  166951262   tennessee     usa    6 days left  
2  silver    3c4pdcgg5jt346413  167655728     georgia     usa    2 days left  
3    blue    1ftfw1et4efc23745  167753855    virginia     usa  22 hours left  
4     red    3gcpcrec2jg473991  167763266     florida     usa  22 hours left

*There is an unnamed column which is not necessary and can be removed*

In [59]:
car_df.drop(['Unnamed: 0'], axis =1, inplace= True)

In [60]:
car_df.head(5)

price      brand    model  year   title_status   mileage   color  \
0   6300     toyota  cruiser  2008  clean vehicle  274117.0   black   
1   2899       ford       se  2011  clean vehicle  190552.0  silver   
2   5350      dodge      mpv  2018  clean vehicle   39590.0  silver   
3  25000       ford     door  2014  clean vehicle   64146.0    blue   
4  27700  chevrolet     1500  2018  clean vehicle    6654.0     red   

                   vin        lot       state country      condition  
0    jtezu11f88k007763  159348797  new jersey     usa   10 days left  
1    2fmdk3gc4bbb02217  166951262   tennessee     usa    6 days left  
2    3c4pdcgg5jt346413  167655728     georgia     usa    2 days left  
3    1ftfw1et4efc23745  167753855    virginia     usa  22 hours left  
4    3gcpcrec2jg473991  167763266     florida     usa  22 hours left

In [61]:
car_df.shape

(2499, 12)

In [62]:
car_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2499 entries, 0 to 2498
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   price         2499 non-null   int64  
 1   brand         2499 non-null   object 
 2   model         2499 non-null   object 
 3   year          2499 non-null   int64  
 4   title_status  2499 non-null   object 
 5   mileage       2499 non-null   float64
 6   color         2499 non-null   object 
 7   vin           2499 non-null   object 
 8   lot           2499 non-null   int64  
 9   state         2499 non-null   object 
 10  country       2499 non-null   object 
 11  condition     2499 non-null   object 
dtypes: float64(1), int64(3), object(8)
memory usage: 234.4+ KB


### *Data Cleaning*

In [63]:
#Check for column wise null value*
car_df.isnull().sum()

price           0
brand           0
model           0
year            0
title_status    0
mileage         0
color           0
vin             0
lot             0
state           0
country         0
condition       0
dtype: int64

In [64]:
#Check for any rows which has all values as null*
car_df.isnull().all(axis=1).sum()

0

In [65]:
round(100*(car_df.isnull().sum()/len(car_df.index)),2)

price           0.0
brand           0.0
model           0.0
year            0.0
title_status    0.0
mileage         0.0
color           0.0
vin             0.0
lot             0.0
state           0.0
country         0.0
condition       0.0
dtype: float64

*As we can see none of the column has null percentage greater than 0. Hence the data looks good*

### *Drop Unnecessary Columns*
*Drop the columns which might not be required for analysis. Here `vin` num and `lot` num are not required*

In [66]:
car_df.head()

price      brand    model  year   title_status   mileage   color  \
0   6300     toyota  cruiser  2008  clean vehicle  274117.0   black   
1   2899       ford       se  2011  clean vehicle  190552.0  silver   
2   5350      dodge      mpv  2018  clean vehicle   39590.0  silver   
3  25000       ford     door  2014  clean vehicle   64146.0    blue   
4  27700  chevrolet     1500  2018  clean vehicle    6654.0     red   

                   vin        lot       state country      condition  
0    jtezu11f88k007763  159348797  new jersey     usa   10 days left  
1    2fmdk3gc4bbb02217  166951262   tennessee     usa    6 days left  
2    3c4pdcgg5jt346413  167655728     georgia     usa    2 days left  
3    1ftfw1et4efc23745  167753855    virginia     usa  22 hours left  
4    3gcpcrec2jg473991  167763266     florida     usa  22 hours left

In [67]:
car_df.drop(['vin','lot'], axis=1, inplace=True)

In [68]:
car_df.head()

price      brand    model  year   title_status   mileage   color  \
0   6300     toyota  cruiser  2008  clean vehicle  274117.0   black   
1   2899       ford       se  2011  clean vehicle  190552.0  silver   
2   5350      dodge      mpv  2018  clean vehicle   39590.0  silver   
3  25000       ford     door  2014  clean vehicle   64146.0    blue   
4  27700  chevrolet     1500  2018  clean vehicle    6654.0     red   

        state country      condition  
0  new jersey     usa   10 days left  
1   tennessee     usa    6 days left  
2     georgia     usa    2 days left  
3    virginia     usa  22 hours left  
4     florida     usa  22 hours left

In [69]:
car_df.describe()

price         year       mileage
count   2499.000000  2499.000000  2.499000e+03
mean   18767.671469  2016.714286  5.229869e+04
std    12116.094936     3.442656  5.970552e+04
min        0.000000  1973.000000  0.000000e+00
25%    10200.000000  2016.000000  2.146650e+04
50%    16900.000000  2018.000000  3.536500e+04
75%    25555.500000  2019.000000  6.347250e+04
max    84900.000000  2020.000000  1.017936e+06

*As the data frame displays the price of some of the cars is 0. Those data are not valid. Hence remove the rows with `price` 0*

In [70]:
car_df = car_df[car_df['price'] !=0]

In [71]:
car_df

price      brand    model  year   title_status   mileage   color  \
0      6300     toyota  cruiser  2008  clean vehicle  274117.0   black   
1      2899       ford       se  2011  clean vehicle  190552.0  silver   
2      5350      dodge      mpv  2018  clean vehicle   39590.0  silver   
3     25000       ford     door  2014  clean vehicle   64146.0    blue   
4     27700  chevrolet     1500  2018  clean vehicle    6654.0     red   
...     ...        ...      ...   ...            ...       ...     ...   
2494   7800     nissan    versa  2019  clean vehicle   23609.0     red   
2495   9200     nissan    versa  2018  clean vehicle   34553.0  silver   
2496   9200     nissan    versa  2018  clean vehicle   31594.0  silver   
2497   9200     nissan    versa  2018  clean vehicle   32557.0   black   
2498   9200     nissan    versa  2018  clean vehicle   31371.0  silver   

           state country      condition  
0     new jersey     usa   10 days left  
1      tennessee     usa    6 days left  
2        georgia     usa    2 days left  
3       virginia     usa  22 hours left  
4        florida     usa  22 hours left  
...          ...     ...            ...  
2494  california     usa    1 days left  
2495     florida     usa  21 hours left  
2496     florida     usa  21 hours left  
2497     florida     usa    2 days left  
2498     florida     usa  21 hours left  

[2456 rows x 10 columns]

*At first we had 2499 rows, now we got 2456 rows. This says around 50 car had price mentioned 0*

### *Data Analysis*
*Since the data is completely clean with no missing values, the analysis can be done to answer the questions*

### *Which brand is more popular?*

In [72]:
Top10brand_bysales = car_df.groupby('brand').model.count().sort_values(ascending=False).head(10)

In [73]:
Top10brand_bysales

brand
ford         1217
dodge         430
nissan        312
chevrolet     286
gmc            40
jeep           30
chrysler       18
bmw            16
hyundai        15
kia            12
Name: model, dtype: int64

*As per the result, 'Ford' has got more cars for sales, hence it is popular*

### *As we know Ford is most popular brand, let's find out which state has the ford price cheapest*

In [89]:
ford = car_df[car_df['brand']=='ford'].groupby(['state','brand']).agg(avg_price=pd.NamedAgg(column='price',aggfunc='mean')).sort_values(by='avg_price',ascending=True).head(5)

In [90]:
ford

avg_price
state    brand              
montana  ford    2825.000000
arkansas ford    5805.000000
georgia  ford   10395.000000
idaho    ford   11500.000000
oregon   ford   11554.761905

*It is clear that `Montana` has the lowest selling price for ford cars*

### *Which brand has more cars of highest price?*

In [91]:
Top_brand_avg_price = car_df.groupby('brand').price.mean().sort_values(ascending=False).head(10)
Top_brand_avg_price

brand
harley-davidson    54680.000000
lincoln            36300.000000
lexus              33220.000000
maserati           30300.000000
land               28900.000000
mercedes-benz      28704.000000
bmw                28046.875000
cadillac           27712.222222
ford               21987.351684
buick              21358.750000
Name: price, dtype: float64

*By Brand, Harley-Davidson has cars with highest price*

### *Which State sell more cars*

In [92]:
highest_selling_state = car_df.groupby('state').model.count().sort_values(ascending=False).head(10)
highest_selling_state

state
pennsylvania      298
florida           245
texas             209
california        185
michigan          169
north carolina    146
minnesota         116
illinois          109
wisconsin          93
virginia           88
Name: model, dtype: int64

*Pennsylvania sells more cars*

### *Which brand has more expensive cars?*

In [97]:
expensive_cars = car_df.sort_values(by= 'price', ascending=False).head(1)

In [98]:
expensive_cars

price          brand     model  year   title_status  mileage   color  \
502  84900  mercedes-benz  sl-class  2017  clean vehicle  25302.0  silver   

       state country    condition  
502  florida     usa  2 days left

*Hence the most expensive car is from `mercedes-benz'*